File: frontend/src/types/schedule.ts
/** 予定データ（REST APIのScheduleDtoに対応） */
export interface Schedule {
  id: number | null;
  title: string;
  isAllDay: boolean;
  startDatetime: string;   // 形式: yyyy/MM/dd-HH:mm
  endDatetime: string;     // 形式: yyyy/MM/dd-HH:mm
  owner: string;
  description: string;
  /** 他のユーザーと共有するかどうか（デフォルト true） */
  shared: boolean;
  /** 作成/更新日時（サーバー側で自動設定、新規作成時は省略可） */
  updateTime?: string;     // 形式: yyyy/MM/dd-HH:mm:ss
  /** オーナーのチップ背景色（API応答用） */
  ownerChipBgColor?: string;
  /** オーナーの表示名（API応答用） */
  ownerDisplayName?: string;
  /** メンバーのユーザー名一覧 */
  memberUsernames?: string[];
  /** メンバーごとのチップ背景色（username → hexColor） */
  memberChipBgColors?: Record<string, string>;
  /** メンバーごとの表示名（username → displayName） */
  memberDisplayNames?: Record<string, string>;
  /** グループID一覧（グループ予定の場合のみ設定） */
  groupIds?: number[];
  /** 現在のユーザーが編集可能か */
  canEdit?: boolean;
}

/** メンバーデータ（GET/POST /api/v1/schedules/{id}/members のレスポンス） */
export interface ScheduleMember {
  id: number;
  scheduleId: number;
  username: string;
  createdAt: string;
}

File: src/main/java/com/rogawa/secretary/domain/model/Schedule.java
package com.rogawa.secretary.domain.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import lombok.Data;

@Data
public class Schedule {
    private Long id;
    private String title;
    private Boolean isAllDay;
    private LocalDateTime startDatetime;
    private LocalDateTime endDatetime;
    private String owner;
    private String description;
    private LocalDateTime updateTime;
    /** 他のユーザーと共有するかどうか（デフォルト true） */
    private Boolean shared = true;

    /** 所属グループID一覧（空=個人予定） */
    private List<Long> groupIds = new ArrayList<>();

    /** オーナーのチップ背景色（API応答用、永続化しない） */
    private String ownerChipBgColor;

    /** オーナーの表示名（API応答用、永続化しない） */
    private String ownerDisplayName;

    /** メンバーのユーザー名一覧（API応答用、永続化しない） */
    private List<String> memberUsernames;

    /** メンバーごとのチップ背景色（API応答用、永続化しない） */
    private Map<String, String> memberChipBgColors;

    /** メンバーごとの表示名（API応答用、永続化しない） */
    private Map<String, String> memberDisplayNames;

    /** 現在のユーザーが編集可能か（API応答用、永続化しない） */
    private Boolean canEdit;
}
